### PR TITLE
Add watch option to elf-graphql cli

### PR DESCRIPTION
**Pull request by ryangoree**
- Added a `--watch`/`-w` option to the `elf-graphql` cli
- Added a new watch script which accepts a glob path and handler fn, then:
  - Finds the closest parent directory it can recursively listen to for changes based on the glob path.
  - Adds a listener to the directories matching the glob path with a debounce
  - Calls the handler fn if a file matching the glob is changed.
- Added a check for the `-w` option in the codegen script which calls `watch` with a glob path for any graphql file and a handler which reruns the codegen.

### CODE REVIEW

File: packages/graphql/src/codegen/generate.ts [hunk 1]
```diff
@@ -13,6 +14,8 @@ interface GenerateOptions {
   outDir: string;
   package?: boolean;
   schema?: string;
+  // alias for `watch` option used to avoid collision with `watch` import
+  w?: boolean;
```
Comment (DannyDelott) [inline packages/graphql/src/codegen/generate.ts:18]: Hm, can we rename the import instead?

```ts
import { watch as watchFn } from "./watch";
```

File: packages/graphql/src/codegen/watch.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import minimatch from "minimatch";
+
+/**
+ * A function to watch a given path and call a handler whenever a change is made.
+ * @param path A path to watch for changes. The path can be a file, directory,
+ *   or glob. Globs are matched using [minimatch](https://github.com/isaacs/minimatch).
+ * @param handler The handler to call each time a file matching the path is
+ *   changed.
+ */
+export function watch(path: string, handler: (filename: string) => void): void {
```
Comment (DannyDelott) [inline packages/graphql/src/codegen/watch.ts:13]: nice, I wonder if we could use `gaze` for this: https://www.npmjs.com/package/gaze

**Review by DannyDelott — APPROVED** [review]
Looks good!

File: packages/graphql/src/codegen/watch.ts (hunk 2 above)
Comment (ryangoree) [reply packages/graphql/src/codegen/watch.ts:13]: Captured this is CU as a todo for possible improvement. https://app.clickup.com/t/2kxz33f

**Comment by ryangoree** [comment]
Rebased

**Pull request merged**